zos-utils/zossrc/javaUsermap.c

```diff
@@ -0,0 +1,122 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+
+#define _OPEN_SYS
+#include <unistd.h>
+#include <errno.h>
+#include "zowe-common-c/h/rusermap.h"
+#include "javaUsermap.h"
+#include <stdio.h>
+
+
+#pragma convert(819)
+const char *JNI_CLASS_ILLEGAL_ARGUMENT_EXCEPTION = "java/lang/IllegalArgumentException";
+
+const char *JNI_SIGNATURE_METHOD_STRING_INT_INT_INT_INT_VOID = "(Ljava/lang/String;IIII)V";
+const char *JNI_SIGNATURE_METHOD_STRING_INT_INT_INT_VOID = "(Ljava/lang/String;III)V";
+const char *JNI_METHOD_CONSTRUCTOR = "<init>";
+
+const char *JNI_CLASS_MAPPER_RESPONSE = "org/zowe/commons/usermap/MapperResponse";
+const char *JNI_CLASS_CERTIFICATE_RESPONSE = "org/zowe/commons/usermap/CertificateResponse";
+
+const char *JNI_MESSAGE_CANNOT_CONVERT_USER_ID = "Cannot convert userID";
+#pragma convert(0)
+
+int strnlen(char *txt, int max) {
+    if (max < 0) return 0;
+    for (int i = 0; i < max; i++) {
+        if (!txt[i]) return i;
+    }
+    return max;
+}
+/**
+ * It reads String from memory in EBCDIC with length up to value in argument length
+ */
+jstring get_jstring(JNIEnv *env, char* ebcdic, int length)
+{
+    if (!ebcdic || (length < 0)) return NULL;
+
+    int realSize = strnlen(ebcdic, length);
+    if (realSize < length) {
+        length = realSize;
+    }
+    char *output = (char*) __malloc31(length + 1);
+    strncpy(output, ebcdic, length);
+    output[length] = 0;
+
+    int size = __etoa(output);
+    if (size < 0) {
+        jclass exception_clazz = (*env) -> FindClass(env, JNI_CLASS_ILLEGAL_ARGUMENT_EXCEPTION);
```

Comment:
This could be done in onLoad method and then just reused (see https://github.com/zowe/common-java/blob/d253350f8232f0e60e711689b6d8e3ffb24f74f4/attls/zossrc/AttlsContext.c#L319)